packages/kbn-config/src/deprecation/apply_deprecations.ts

```diff
@@ -22,9 +23,21 @@ export const applyDeprecations = (
   deprecations: ConfigDeprecationWithContext[],
   createAddDeprecation: (pluginId: string) => AddConfigDeprecation = noopAddDeprecationFactory
 ) => {
-  let processed = cloneDeep(config);
+  const result = cloneDeep(config);
   deprecations.forEach(({ deprecation, path }) => {
-    processed = deprecation(processed, path, createAddDeprecation(path));
+    const commands = deprecation(result, path, createAddDeprecation(path));
+    if (commands) {
+      if (commands.set) {
+        commands.set.forEach(function ({ key, value }) {
+          set(result, key, value);
+        });
```

Comment:
Any reason to not use arrow func here?

Reply:
nope, but [prefer-arrow-callback](https://eslint.org/docs/rules/prefer-arrow-callback) eslint is not enabled in the repo